### PR TITLE
Mldb 1525 rowname generator explain

### PR DESCRIPTION
**Pull request by steadtler**
Add a new field to GenerateRowsWhereFunction to track its complexity so that we can reason on it.

### CODE REVIEW

File: sql/sql_expression.h [hunk 1]
```diff
@@ -1301,14 +1301,24 @@ PREDECLARE_VALUE_DESCRIPTION(TupleExpression);
 */
 struct GenerateRowsWhereFunction {
 
+    enum Complexity {        
+       
+        CONSTANT = 0,
+        BETTER_THAN_TABLESCAN,        
+        UNFILTERED_TABLESCAN,
```
Comment (guyd) [inline sql/sql_expression.h]: What would be an example of `UNFILTERED_TABLESCAN`? I could not find any example in the test case.

**Comment by guyd** [comment]
+1

File: sql/sql_expression.h (hunk 1 above)
Comment (steadtler) [inline sql/sql_expression.h]: When you provide no WHERE clause, or some WHERE clause that evaluate to always true. Basically anytime we want all rows.

Comment (guyd) [inline sql/sql_expression.h]: So how is it different than TABLESCAN then?

**Pull request merged**